SharedSources/MediaLibraryModel/VideoModel.swift

```diff
@@ -95,12 +95,25 @@ extension VLCMLMedia {
 }
 
 extension VLCMLMedia {
-    @objc func formatDuration() -> String {
+    @objc func mediaDuration() -> String {
         return String(format: "%@", VLCTime(int: Int32(duration())))
     }
 
     @objc func formatSize() -> String {
         return ByteCountFormatter.string(fromByteCount: Int64(mainFile().size()),
                                          countStyle: .file)
     }
+
+    func mediaProgress() -> Float {
+        guard let string = metadata(of: .progress).str as NSString? else {
+            return 0.0
```

Comment:
Is this for safety?

Reply:
when there is no progress yet set the default will be 0